src/services/facilityService.ts

```diff
@@ -19,36 +17,42 @@ export const getFacilityById = async (id: string) : Promise<Facility | null> =>
     return convertedEntity
 }
 
-export const addFacility = async (input: FacilityInput) : Promise<Facility> => {
-    const db = getFirestore()
+export async function addFacility(input: gqlTypes.Facility) {
+    const db = firebase.getFirestore()
    
-    const facilityRef = db.collection('facilities')
-
-    const healthcareProfessionalIds = await mapAndValidateHealthcareProInput(
-        input.healthcareProfessionals as HealthcareProfessionalInput[]
-    )
+    const facilityRef = db.collection('facilities').doc()
+    const healthcareProfessionalRef = db.collection('healthcareProfessionals').doc()
+
+    if (input.healthcareProfessionals !== null 
+        && input.healthcareProfessionals !== undefined 
+        && input.healthcareProfessionals.length > 0) {
+        addHealthcareProfessional(
```

Comment:
we'll want to be able to add more than one. We should do this for the whole array not just the first item. 

you'll probably need an awaited for loop for each item to save...like this
https://stackoverflow.com/a/50874507/3417143

Reply:
@ermish See my previous comment. I couldn't get it working because it would only return a promise. I'd like to keep it as one for now and if you could please refactor it when you jump into the backend work. 🙏🏾

Reply:
Sounds good 👍 I can update in a separate PR